apis/v1alpha1/httproute_types.go

```diff
@@ -145,13 +146,23 @@ type HTTPRouteMatch struct {
 	// +optional
 	// +kubebuilder:validation:Enum=Exact;ImplementationSpecific
 	// +kubebuilder:default=Exact
-	HeaderMatchType *string `json:"headerMatchType" protobuf:"bytes,3,opt,name=headerMatchType"`
+	Type *string `json:"type" protobuf:"bytes,3,opt,name=type"`
```

Comment:
Does this need to be a pointer? It seems like nil and "" are indistinguishable here and we're already setting a default value.

Reply:
I feel this no longer needs to be a pointer. If I recall correctly, having this field be non-pointer and default to `Exact` but avoiding the case where no header match is meant to be set was the initial motivation to make `HTTPHeaderMatch` into its own struct.

Reply:
Yeh, we can remove the pointer, since the defaulting has precedence anyway.